accounts/views.py

```diff
@@ -8,36 +8,34 @@
 from django.shortcuts import redirect
 from django.utils.decorators import method_decorator
 from django.views.decorators.csrf import csrf_exempt
+from django.views.generic.base import View
 from oauth2_provider.models import get_access_token_model
 from oauth2_provider.views import IntrospectTokenView
-from rest_framework import generics
-from rest_framework_api_key.permissions import HasAPIKey
 from sentry_sdk import capture_message
 
 from accounts.auth import LabsView, PennView
 from accounts.serializers import UserSerializer
 
 
-class LoginView(generics.GenericAPIView):
+class LoginView(View):
     """
     Log in a user.
+    WARNING: You must ensure this page is protected by Shibboleth and Clean Headers
+    See https://github.com/nginx-shib/nginx-http-shibboleth
```

Comment:
From a cursory read of the linked page, I'm guessing this means that not everyone can hit the `LoginView` endpoint. I see this is in the `accounts` package... will we have to add a custom nginx rule on every dokku app that uses this package?

Reply:
We just need to make sure that when platform is (re)deployed in production, we configure nginx to require Shibboleth authentication on the `accounts/login/` route. This only needs to be done on platform though.